2020/4/passport/src/main.rs

```diff
@@ -0,0 +1,248 @@
+use std::env;
+use std::fs::File;
+use std::io::{self, BufRead};
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let file = File::open(&args[1]);
+
+    let reader = io::BufReader::new(file.unwrap());
+    let mut current = String::from("");
+    let mut passports = Vec::new();
+    for line in reader.lines() {
```

Comment:
Heh, you could probably make this super functional with [`scan`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.scan) but Rust isn't necessarily "a functional language", kinda like how C# has functional elements. This is still totally idiomatic as-is.